app/components/offers-block.js

```diff
@@ -0,0 +1,40 @@
+import { inject as service } from "@ember/service";
+import { getWithDefault } from "@ember/object";
+import Component from "@ember/component";
+import _ from "lodash";
+
+export default Component.extend({
+  store: service(),
+  locationName: "",
+
+  didReceiveAttrs() {
+    this._super(...arguments);
+    console.log(this.get("record"));
+    const location = this.get("store").peekRecord(
+      "district",
+      this.get("record").district_id
+    );
+    this.getPackageTypesCounts(this.get("record").items);
+    this.set("locationName", getWithDefault(location, "name", "N/A"));
+    this.set("initialItems", this.get("record").items.slice(0, 4));
+  },
```

Comment:
By doing `this.get("record").district_id` and `this.get("record").items`, you're making this component non-generic in the sense that it can't be used with any other data than exactly the one you're fetching. Not a very re-usable component